exp/controllers/azuremachinepool_controller.go

```diff
@@ -293,6 +294,22 @@ func (ampr *AzureMachinePoolReconciler) reconcileNormal(ctx context.Context, mac
 		return reconcile.Result{}, nil
 	}
 
+	var reconcileError azure.ReconcileError
+
+	// Initialize the cache to be used by the AzureMachine services.
+	err := machinePoolScope.InitMachinePoolCache(ctx)
+	if err != nil {
+		if errors.As(err, &reconcileError) && reconcileError.IsTerminal() {
+			ampr.Recorder.Eventf(machinePoolScope.AzureMachinePool, corev1.EventTypeWarning, "SKUNotFound", errors.Wrap(err, "failed to initialize machine cache").Error())
+			log.Error(err, "Failed to initialize machine cache")
+			machinePoolScope.SetFailureReason(capierrors.InvalidConfigurationMachineError)
+			machinePoolScope.SetFailureMessage(err)
+			machinePoolScope.SetNotReady()
+			return reconcile.Result{}, nil
```

Comment:
I adapted this from the `machine_controller` which uses the same logic. I think it is returning nil since we want to retry if the cache fails to initalize.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azuremachine_controller.go#L275

Reply:
Gotcha. I was under the impression if we returned `nil`, it would wait 10 minutes before requeueing, and that we need to return an error to retry. @CecileRobertMichon am I mistaken here?

Reply:
We return nil because we _don't_ want to queue a retry, since it's a terminal error (I believe in this case it's "SKU not found", this happens if the user provides an invalid VM size) it won't succeed on a retry unless the user changes the spec